global_paths/plot_global_path.py

```diff
@@ -0,0 +1,125 @@
+""""This is a script to plot the global path from a csv file.
+
+Command Line Arguments:
+
+    file_path (Str): The relative path to the global path file
+    interval_spacing (float): The distance between waypoints km, default 30.0
+    write (Bool): Whether to write the generated path to a new file
+    pos (Tuple): Starting position in latitude and longitude
+
+usage in local_pathfinding terminal:
+
+python3 global_paths/plot_global_path.py <file_path> --interval_spacing <interval_spacing> --write
+--pos <latitude> <longitude>
+
+
+"""
+import argparse
+import csv
+from typing import Dict, List, Tuple
+
+import plotly.graph_objects as go
+from custom_interfaces.msg import HelperLatLon
+
+from local_pathfinding.node_mock_global_path import MOCK_GPS, MockGlobalPath
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("file_path", help="The relative path to the global path file")
+    parser.add_argument(
+        "-i",
+        "--interval_spacing",
+        type=float,
+        default=30.0,
+        help="The distance between waypoints km, default 30.0",
+    )
+    parser.add_argument("--write", action="store_true", help="Write the path to a new file")
+    parser.add_argument(
+        "--pos",
+        nargs=2,
+        type=float,
+        default=[MOCK_GPS.lat_lon.latitude, MOCK_GPS.lat_lon.longitude],
+        help="Starting position",
+    )
+
+    args = parser.parse_args()
+
+    lats, lons = get_lats_and_lons(args.file_path, args.interval_spacing, args.pos, args.write)
+    plot_global_path(lats, lons)
+    print("global path:", lats_and_lons_to_dict(lats, lons), sep="\n")
+
+
+def get_lats_and_lons(
+    file_path: str, interval_spacing: float, pos: tuple, write: bool
+) -> Tuple[List[float], List[float]]:
+    lats = []
+    lons = []
+
+    with open(file_path, "r") as file:
+        reader = csv.reader(file)
+        # skip header
+        reader.__next__()
+        for row in reader:
+            lats.append(float(row[0]))
+            lons.append(float(row[1]))
+
+    if len(lats) < 2:
```

Comment:
If this is supposed to reflect what happens in `global_path_callback`, need to add the case where the waypoints are too far apart. But to reduce complexity maybe we can remove this if statement and only plot the waypoints in the file. Creating the path can be done by path_builder

Reply:
I like that. I have removed the if, and this function will just get the lats an lons from a file and return them.